src/radical/pilot/agent/launch_method/mpirun.py

```diff
@@ -15,114 +15,235 @@ class MPIRun(LaunchMethod):
 
     # --------------------------------------------------------------------------
     #
-    def __init__(self, name, cfg, session):
+    def __init__(self, name, cfg, log, prof):
 
-        self._mpt    = False
-        self._rsh    = False
-        self._ccmrun = ''
-        self._dplace = ''
+        self._mpt      = False
+        self._rsh      = False
+        self._ccmrun   = ''
+        self._dplace   = ''
+        self._info     = None
 
-        LaunchMethod.__init__(self, name, cfg, session)
+        self._env_orig = ru.env_eval('env/orig.env')
 
+        log.debug('===== lm MPIRUN init start')
+
+        LaunchMethod.__init__(self, name, cfg, log, prof)
+
+        self._init_from_info()
+        self._log.debug('===== lm MPIRUN init stop')
+
+
+    # --------------------------------------------------------------------------
+    #
+    def _init_from_info(self):
+
+        self._log.debug('===== lm MPIRUN init_info start')
+
+        if self._info:
+
+            self._log.debug('===== lm MPIRUN init_info active: %s', self._info)
+
+            self._command     = self._info['command']
+            self._dplace      = self._info['dplace']
+            self._ccmrun      = self._info['ccmrun']
+            self._mpt         = self._info['mpt']
+            self._rsh         = self._info['rsh']
+            self._mpi_version = self._info['mpi_version']
+            self._mpi_flavor  = self._info['mpi_flavor']
+            self._env_sh      = self._info['env_sh']
+
+            self._env         = ru.env_eval(self._env_sh)
+
+            self._log.debug('===== lm MPIRUN init_info stop: %s', self._command)
+
+        else:
+            self._log.debug('===== lm MPIRUN init_info stop: ---')
 
     # --------------------------------------------------------------------------
     #
-    def _configure(self):
+    def initialize(self, rm, lmcfg):
+
+        self._log.debug('===== lm MPIRUN initialize start')
+
+        if self._info:
+            raise RuntimeError('LM %s already initialized' % self.name)
+
+        command     = None
+        dplace      = ''
+        ccmrun      = ''
+        mpt         = False
+        rsh         = False
+        mpi_version = None
+        mpi_flavor  = None
+        env_sh      = '%s/env/%s.env' % (self._pwd, self.name)
+
+        pre_exec  = lmcfg.get('pre_exec', [])
+        cmds      = lmcfg.get('cmds',     [])
+
+        # set up LM environment: first recreate the agent's original env, then
+        # apply the launch method's env_prep commands.  Store the resulting env
+        # in `env/lm_<name>.env`
```

Comment:
`env/<name>.env`?

Reply:
I want to avoid collisions with other env names, thus the prefix.